builtin/bisect--helper.c

```diff
@@ -431,14 +646,17 @@ int cmd_bisect__helper(int argc, const char **argv, const char *prefix)
 			 N_("check whether bad or good terms exist"), BISECT_NEXT_CHECK),
 		OPT_CMDMODE(0, "bisect-terms", &cmdmode,
 			 N_("print out the bisect terms"), BISECT_TERMS),
+		OPT_CMDMODE(0, "bisect-start", &cmdmode,
+			 N_("start the bisect session"), BISECT_START),
 		OPT_BOOL(0, "no-checkout", &no_checkout,
 			 N_("update BISECT_HEAD instead of checking out the current commit")),
 		OPT_END()
 	};
 	struct bisect_terms terms = { .term_good = NULL, .term_bad = NULL };
 
 	argc = parse_options(argc, argv, prefix, options,
-			     git_bisect_helper_usage, PARSE_OPT_KEEP_UNKNOWN);
+			     git_bisect_helper_usage,
+			     PARSE_OPT_KEEP_DASHDASH | PARSE_OPT_KEEP_UNKNOWN);
```

Comment:
Maybe we should talk about this in the commit message. If this was not `bisect--helper` but instead `bisect`, I would expect regular user input to be parsed here, and I would suspect that we would have to handle the `--` separator specifically in quite a few places after this change. But since the `bisect--helper` is intended to be called from the `git bisect` script, I think it is safe to say that only known call sites call this, with little in the way of arbitrary user input.

Reply:
Can you please elaborate on this?

Reply:
When we keep a `--` in the `argv`, then we typically need to handle this in *all* subsequent code paths. However, in this instance, we are talking about a helper command that is not supposed to be called by the Git user directly, but via the `git bisect` script. That means that the helper will receive only specifically-crafted command-line parameters, and therefore we can be a little lax with the `--` and not teach each and every possible code path after the `parse_options()` call to all of a sudden expect a `--` to be a valid item in the `argv` array.

Reply:
So, what should be the commit message like?

Reply:
I would simply add a paragraph along those lines:

> Note that we are a bit lax about command-line parsing because the helper is not supposed to be called by the user directly (but only from the `git bisect` script).